src/app/main/list-news/list-news.component.ts

```diff
@@ -0,0 +1,99 @@
+import { Component, OnInit, OnDestroy } from '@angular/core';
+import { DataService } from '../../services/data.service';
+import { News } from '../../interfaces/news.interface';
+import { FormControl, FormBuilder, FormGroup } from '@angular/forms';
+import { Subscription } from 'rxjs';
+
+@Component({
+  selector: 'app-list-news',
+  templateUrl: './list-news.component.html',
+  styleUrls: ['./list-news.component.scss']
+})
+export class ListNewsComponent implements OnInit, OnDestroy {
+
+  public newsItems: News[];
+  public currentItems: News[] = [];
+  public isEmpty: boolean = false;
+
+  public filter = new FormGroup({
+    keyWords: new FormControl()
+  });
+
+  public source: string = null;
+  public showSource: string = null;
+  public isDisabled: boolean = false;
+
+  protected readonly subscriptions: Subscription[] = [];
+
+
+  constructor(private fb: FormBuilder,
+              private dataService: DataService) { }
+
+  public ngOnInit(): void {
+      this.dataService.currentNews.subscribe((data: News[]) => {
+
+          if (data.length !== 10) {
+            this.newsItems = data.slice(0, 5);
+          } else {
+            this.newsItems = data;
+          }
+
+          if (data.length < 5) {
+            this.isEmpty = true;
+          } else {
+            this.isEmpty = false;
+          }
+      });
+
+      this.filter = this.fb.group({
+        keyWords: null,
+      });
+  }
+
+  public ngOnDestroy(): void {
+    this.subscriptions.forEach(subscription => subscription.unsubscribe());
+  }
+
+  public loadNews(event: any): void {
+    this.isEmpty = true;
+  }
+
+  public checkValue(event: any) {
+    if (event.target.checked) {
+      this.isDisabled = true;
+      this.dataService.filterByMe();
+    } else {
+      this.isDisabled = false;
+      this.dataService.getAllItems();
+    }
+  }
+
+  public filterSource(event: any): void {
+    if (event.target.value !== 'Select source') {
+      this.source = event.target.value;
```

Comment:
create variable and assign event.target.value and reuse this variable. No need to extract alwayss value